keyboards/redox/proton_c/config.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 xerootg
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4D44 // "MD"
+#define PRODUCT_ID      0x5244 // "RD"
+#define DEVICE_VER      0x0101 // added 1 for the rev1 + proton c
+#define MANUFACTURER    Falbatech
+#define PRODUCT         The Redox Keyboard
+
+/* key matrix size */
+// Rows are doubled-up
+#define MATRIX_ROWS 10
+#define MATRIX_COLS 7
+
+// wiring of each half
+#define MATRIX_ROW_PINS { B5, B3, B2, B1, B0 }
+#define MATRIX_COL_PINS { A1, A0, B8, B13, B14, B15, B9 }
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6, F5} //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
+// #define BACKLIGHT_LEVELS 3
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* serial.c configuration for split keyboard */
+#define SERIAL_USART_FULL_DUPLEX   // Enable full duplex operation mode.
+#define SERIAL_USART_TX_PIN B6     // USART TX pin
+#define SERIAL_USART_RX_PIN B7     // USART RX pin
+// #define USART1_REMAP             // Remap USART TX and RX pins on STM32F103 MCUs, see table below.
+// #define SERIAL_USART_PIN_SWAP    // Swap TX and RX pins if keyboard is master halve.
```

Comment:
I would expect that the full duplex configuration for this board needs `#define SERIAL_USART_PIN_SWAP`, because you are reusing the wiring originally intended for I2C.  Are you handling the TX/RX pin swap in hardware instead?

Reply:
It was unclear to me how to do this correctly. On my keyboard, I compiled one half with that set and the other half, not. I did see that there's a provision in the code to identify the left vs right half via a pin state, but I figured that would be a hardware modification. Is there a better way?

Reply:
`#define SERIAL_USART_PIN_SWAP` should affect only the master half (does not matter whether it is right or left — it would be whatever half that has USB connected), so you should just add that option and flash the same firmware into both halves.

Reply:
I'll push the change, I'll have to test it before I resolve this one I suppose.

Reply:
Ok, tested on the keyboard. This does indeed work on both halves.